src/components/CIPipelineN/AdvancedConfigOptions.tsx

```diff
@@ -0,0 +1,184 @@
+    const [collapsedSection, setCollapsedSection] = useState<boolean>(false)
+    const [allowOverride, setAllowOverride] = useState<boolean>(ciPipeline?.isDockerConfigOverridden ?? false)
+    const [parentState, setParentState] = useState<{
+        loadingState: ComponentStates
+        selectedCIPipeline: CIPipelineDataType
+        dockerRegistries: any
+        sourceConfig: any
+        ciConfig: CiPipelineResult
+        defaultDockerConfigs: DockerConfigOverrideType
+    }>({
+        loadingState: ComponentStates.loading,
+        selectedCIPipeline: ciPipeline,
+        dockerRegistries: null,
+        sourceConfig: null,
+        ciConfig: null,
+        defaultDockerConfigs: null,
+    })
+
+    const addDockerArg = (): void => {
+        const _form = { ...formData }
+        if (_form.args > 0) {
+            _form.args.unshift({ key: '', value: '' })
+        } else {
+            _form.args.push({ key: '', value: '' })
+        }
+        setFormData(_form)
+    }
+
+    const handleDockerArgChange = (event, index: number, key: 'key' | 'value'): void => {
+        const _form = { ...formData }
+        _form.args[index][key] = event.target.value
+        setFormData(_form)
+    }
+
+    const removeDockerArgs = (index: number): void => {
+        const _form = { ...formData }
+        const newArgs = []
+        for (let i = 0; i < _form.args.length; i++) {
+            if (index != i) newArgs.push(_form.args[i])
+        }
+        _form.args = newArgs
+        setFormData(_form)
+    }
+
+    const updateDockerConfigOverride = (key: string, value: any) => {
+        const _form = {
+            isDockerConfigOverridden: ciPipeline?.isDockerConfigOverridden,
+            ...formData,
+        }
+        const keyPair = key.split('.')
+
+        // Init the dockerConfigOverride with global values if dockerConfigOverride data is not present
+        if (
+            !ciPipeline?.isDockerConfigOverridden &&
+            (!formData.dockerConfigOverride || !Object.keys(formData.dockerConfigOverride).length) &&
+            parentState?.ciConfig
+        ) {
+            _form.dockerConfigOverride = {
+                dockerRegistry: parentState.ciConfig.dockerRegistry,
+                dockerRepository: parentState.ciConfig.dockerRepository,
+                dockerBuildConfig: {
+                    gitMaterialId: parentState.ciConfig.dockerBuildConfig.gitMaterialId,
+                    dockerfileRelativePath: parentState.ciConfig.dockerBuildConfig.dockerfileRelativePath,
+                },
+            }
+        }
+
+        if (key.startsWith('dockerConfigOverride.dockerBuildConfig')) {
+            _form[keyPair[0]][keyPair[1]][keyPair[2]] = value
+        } else if (key.startsWith('dockerConfigOverride')) {
+            _form[keyPair[0]][keyPair[1]] = value
+        } else if (key === 'isDockerConfigOverridden') {
```

Comment:
Please use constant for `isDockerConfigOverridden`

Reply:
It's not going to be used anywhere else. Just a single-time check.